ReactWindows/ReactNative.Shared/Bridge/PromiseExtensions.cs

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Portions derived from React Native:
+// Copyright (c) 2015-present, Facebook, Inc.
+// Licensed under the MIT License.
+
+using Newtonsoft.Json.Linq;
+using System;
+
+namespace ReactNative.Bridge
+{
+    /// <summary>
+    /// Extension methods for promised.
+    /// </summary>
+    public static class PromiseExtensions
+    {
+        private const string DefaultError = "EUNSPECIFIED";
+
+        /// <summary>
+        /// Report an error which wasn't caused by an exception.
+        /// </summary>
+        /// <param name="promise">The promise.</param>
+        /// <param name="code">The error code.</param>
+        /// <param name="message">The error message.</param>
+        public static void Reject(this IPromise promise, string code, string message)
```

Comment:
I love this pattern of slimming an interface down to a cupcake and using extension methods as decoupled icing.